app/scripts/lib/dapp-swap/dapp-swap-util.ts

```diff
@@ -0,0 +1,121 @@
+import log from 'loglevel';
+import { Hex, JsonRpcParams, JsonRpcRequest } from '@metamask/utils';
+import {
+  GenericQuoteRequest,
+  QuoteResponse,
+} from '@metamask/bridge-controller';
+import {
+  NetworkClientId,
+  NetworkConfiguration,
+} from '@metamask/network-controller';
+
+import { captureException } from '../../../../shared/lib/sentry';
+import { getDataFromSwap } from '../../../../shared/modules/dapp-swap-comparison/dapp-swap-comparison-utils';
+import { SecurityAlertResponse } from '../ppom/types';
+
+export type DappSwapMiddlewareRequest<
+  Params extends JsonRpcParams = JsonRpcParams,
+> = Required<JsonRpcRequest<Params>> & {
+  origin?: string;
+  securityAlertResponse?: SecurityAlertResponse | undefined;
+  networkClientId: NetworkClientId;
+  params: {
+    data: string;
+    from: string;
+    chainId: string;
+    calls: { data: string; from: string }[];
+  }[];
+};
+
+const FOUR_BYTE_EXECUTE_SWAP_CONTRACT = '0x3593564c';
+const DAPP_SWAP_COMPARISON_ORIGIN = 'https://app.uniswap.org';
+const TEST_DAPP_ORIGIN = 'https://metamask.github.io';
+const DEFAULT_QUOTEFEE = 250;
```

Comment:
Do we need a default one, if we are already reading it from remote flags?

Reply:
I left it as default, but yep can be removed. Let me update the PR.

Reply:
PR is updated